napari/utils/indexing.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+
+
+def index_in_slice(index, position_in_axes):
+    """Convert a NumPy fancy indexing expression from data to sliced space.
+
+    Parameters
+    ----------
+    index : tuple of array of int
+        A NumPy fancy indexing expression [1]_.
+    position_in_axes : dict[int, int]
+        A dictionary mapping sliced (non-displayed) axes to a slice position.
+
+    Returns
+    -------
+    sliced_index : tuple of array of int
+        The indexing expression (nD) restricted to the current slice (usually
+        2D or 3D).
+
+    Examples
+    --------
+    >>> index = (np.arange(5), np.full(5, 1), np.arange(4, 9))
+    >>> index_in_slice(index, {0: 3})
+    (array([1]), array([7]))
+    >>> index_in_slice(index, {1: 1, 2: 8})
+    (array([4]),)
```

Comment:
It looks like a good material for unit test.

Reply:
Personally I'm a big fan of doctests. Do we run doctests? I hope so!

Reply:
It is not so easy setup doctest (for example you do not import numpy here).

Reply:
Doctests are run with module scope, so `np` is available because it's imported in the module...